src/components/tests/services/tests.service.js

```diff
@@ -0,0 +1,156 @@
+                    var exerciseId;
+                    for (var i = 0, ii = arr.length; i < ii; i++) {
+                        if (arr[i].categoryId === id) {
+                            exerciseId = arr[i].id;
+                            break;
+                        }
+                    }
+                    return exerciseId;
+                };
+
+                TestsSrv.getExamsAndExerciseResults = function (examCopy, examResults) {
+                    var examResult = examResults[0];
+                    var sectionResults = Object.keys(examResult.sectionResults);
+                    var exerciseResultsProms = [];
+                    var examSectionProms = [];
+                    angular.forEach(sectionResults, function (sectionId) {
+                        sectionId = +sectionId;
+                        exerciseResultsProms.push(ExerciseResultSrv.getExerciseResult(ExerciseTypeEnum.SECTION.enum, sectionId, examCopy.id, null, true));
+                        examSectionProms.push(TestsSrv.getExamSection(examCopy, sectionId));
+                    });
+                    var getAllExerciseResultsProm = $q.all(exerciseResultsProms);
+                    var getAllExamSectionsProm = $q.all(examSectionProms);
+                    return $q.all([getAllExerciseResultsProm, getAllExamSectionsProm]).then(function (res) {
+                        var exerciseResults = res[0];
+                        var examSection = res[1];
+                        return {
+                            exerciseResults: exerciseResults,
+                            examSection: examSection
+                        };
+                    });
+                };
+
+                var subjectOrderProm = TestsSrv.getSubjectsMap();
+
+                TestsSrv.getFullExamSubAndCrossScores = function (sections, sectionsResults) {
+                    $q.when(subjectOrderProm).then(function (res) {
+                        var subjectsOrder = res;
+                        var essayEnum;
+                        for (var i = 0; i < subjectsOrder.subjects.length; i++) {
+                            if (subjectsOrder.subjects[i].id === 8) {
+                                essayEnum = subjectsOrder.subjects[i].id;
+                            }
+                        }
+                        var essayExerciseId = TestsSrv.getExerciseIdByCategoryId(sections, essayEnum);
+                        var newSections = TestsSrv.filterArrByCategoryId(sections, essayEnum);
+                        var newSectionsResults = sectionsResults.filter(function (sectionResult) {
+                            return +sectionResult.exerciseId !== essayExerciseId;
+                        });
+                        return ScoringService.getFullExamSubAndCrossScores(newSections, newSectionsResults);
+                    });
+                };
+
+                TestsSrv.isTypeFull = function (typeId) {
+                    return ScoringService.isTypeFull(typeId);
```

Comment:
some apps use the subjectEnum and other use examTypeEnum. I thought it would be better if it'll come from outside